sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -133,8 +147,16 @@ private void configureChannelPipelineHandlers() {
             .doOnCancel(() -> {
                 ReactorNettyHttpResponse reactorNettyHttpResponse = responseReference.get();
                 if (reactorNettyHttpResponse != null) {
-                    reactorNettyHttpResponse.releaseAfterCancel(request.httpMethod());
+                    reactorNettyHttpResponse.releaseOnNotSubscribedResponse(ReactorNettyResponseState.CANCELLED);
+                }
+            })
+            .onErrorMap(throwable -> {
+                logger.error("Error occurred while sending request : ", throwable);
+                ReactorNettyHttpResponse reactorNettyHttpResponse = responseReference.get();
+                if (reactorNettyHttpResponse != null) {
+                    reactorNettyHttpResponse.releaseOnNotSubscribedResponse(ReactorNettyResponseState.ERROR);
```

Comment:
this is for IllegalReferenceCountException ?

Reply:
Yes, this is a safety block for `IllegalReferenceCountException` :)